rotkehlchen/externalapis/covalent.py

```diff
@@ -108,7 +112,8 @@ def _query(
         query_str += f'/{module}/'
 
         own_key = self._get_api_key()
-        query_str += f'?key={own_key if own_key else KEY}'
+        covalent_key = random.choice(COVALENT_KEYS)
+        query_str += f'?key={own_key if own_key else covalent_key}'
```

Comment:
no need to always run the randomization in every query

```suggestion
        if own_key is None:
            query_str += f'?key={random.choice(COVALENT_KEYS)}'
        else:
            query_str += f'?key={own_key}'
```